Clean Code/ToDo/Program.cs

```diff
@@ -1,110 +1,156 @@
-            Console.WriteLine("1. New task");
-            Console.WriteLine("2. Remove task");
-            Console.WriteLine("3. Pending tasks");
-            Console.WriteLine("4. Exit");
-
-            // Read line
-            string line = Console.ReadLine();
-            return Convert.ToInt32(line);
+            int userOption = 0;
+
+            try
+            {
+                Console.WriteLine("----------------------------------------");
+                Console.WriteLine("Enter the option to perform: ");
+                Console.WriteLine("1. New task");
+                Console.WriteLine("2. Remove task");
+                Console.WriteLine("3. Pending tasks");
+                Console.WriteLine("4. Exit");
+
+                string? line = Console.ReadLine();
+                userOption = Convert.ToInt32(line);
+
+                if (userOption <= 0 || userOption > 4)
+                {
+                    Console.WriteLine($"Incorrect option.");
+                    return 0;
+                }
+
+                return userOption;
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Incorrect option. ({ex.Message})");
+            }
+
+            return userOption;
         }
 
-        public static void ShowMenuDos()
+        private static void ShowMenuAddTask()
+        {
+            Console.WriteLine("Enter the name of the task: ");
+
+            string? task = Console.ReadLine();
+
+            if (task == null || task?.Length <= 0)
+            {
+                Console.WriteLine("No task was added.");
+                return;
+            }
+
+            TaskList?.Add(task);
+            Console.WriteLine("Task registered successfully");
+        }
+
+        private static void ShowMenuRemoveTask()
         {
             try
             {
+                if (TaskList?.Count <= 0)
+                {
+                    Console.WriteLine("There are no tasks to remove.");
+                    return;
+                }
+
                 Console.WriteLine("Enter the number of the task to remove: ");
                 // Show current taks
-                for (int i = 0; i < TL.Count; i++)
+                for (int i = 0; i < TaskList?.Count; i++)
                 {
-                    Console.WriteLine((i + 1) + ". " + TL[i]);
+                    int j = i + 1;
+
+                    Console.WriteLine(j + ". " + TaskList[i]);
                 }
                 Console.WriteLine("----------------------------------------");
 
-                string line = Console.ReadLine();
+                string? line = Console.ReadLine();
+                if (line?.Length <= 0)
+                {
+                    Console.WriteLine("No task was removed.");
+                    return;
+                }
+
                 // Remove one position
                 int indexToRemove = Convert.ToInt32(line) - 1;
+
                 if (indexToRemove > -1)
```

Comment:
What do you think to refactor these lines like this:
```suggestion
                if (indexToRemove > (TaskList.Count() - 1) || indexToRemove < 0)
                {
                    Console.WriteLine("The task number is invalid");
                }
                else if (indexToRemove > -1 && TaskList.Count > 0)
                {
                    string task = TaskList[indexToRemove];
                    TaskList.RemoveAt(indexToRemove);
                    Console.WriteLine($"Task {task} deleted");
                }
```